### PR TITLE
New API

### PR DESCRIPTION
**Pull request by skalnik**
PurpleAir updated their API! It's new and shiny, and much like the old, but slightly not. This PR updates AQI WTF to use it.

### CODE REVIEW

**Comment by obra** [comment]
Is it Kosher to stick your api key in version control like this? if so, I’m +1 to merge

**Comment by skalnik** [comment]
> Is it Kosher to stick your api key in version control like this? if so, I’m +1 to merge

🤷 This app is all client side so it'll be in accessible to end users easily, no matter what. I'm leaning on yolo yes and see what happens. It is, at least, read only, which helps.

File: app.js [hunk 1]
```diff
@@ -213,24 +192,19 @@
...
-    fields["latitude"] = json.fields.findIndex((e) => e === "Lat");
-    fields["longitude"] = json.fields.findIndex((e) => e === "Lon");
-    fields["id"] = json.fields.findIndex((e) => e === "ID");
-    fields["age"] = json.fields.findIndex((e) => e === "age");
+    fields["latitude"] = json.fields.findIndex((e) => e === "latitude");
+    fields["longitude"] = json.fields.findIndex((e) => e === "longitude");
+    fields["sensor_index"] = json.fields.findIndex((e) => e === "sensor_index");
     for (const sensor of json.data) {
-      // Ignore sensors which are either indoor or updated over 5 minutes ago
-      if (sensor[fields["indoor"]] === 0 && sensor[fields["age"]] < 5) {
```
Comment (skalnik) [inline app.js:223]: This is now done server side, via `location_type=0&max_age=300` appended to the full sensor list API requests 🎉

File: app.js [hunk 2]
```diff
@@ -89,16 +89,20 @@
...
 
   function updateAQI(sensor) {
-    let pm25s = [];
-    let humidity = sensor.results[0].humidity;
-
+    sensor = sensor.sensor;
     announceState("Calculating AQI");
 
-    for (const subsensor of sensor.results) {
-      if (!bustedSensor(subsensor)) {
```
Comment (skalnik) [inline app.js:98]: We get a lot less information from PurpleAir now, making this check a lot simpler, so I brought it inline. If we ever want, we can request more data from them and bring this back.

**Comment by obra** [comment]
works for me.

**Pull request merged**